### PR TITLE
KBC-1726 DPM components tests consolidation part3

### PR DESCRIPTION
**Pull request by jirkasemmler**
Jira: KBC-1726

- Last bulk from https://docs.google.com/spreadsheets/d/1bNYwPVlDZuybG56X1wFT58YMq_8jztMhaFF4Snfhdd8/edit#gid=0
- BranchComponentTest uses `createDevBranchForTestCase`
- setup uses cleanup method

### CODE REVIEW

File: tests/Common/ComponentsTest.php [hunk 1]
```diff
@@ -245,10 +232,11 @@ public function testComponentConfigRenew()
     }
 
     /**
-     * @dataProvider provideComponentsClientType
-     * @param string $clientType
+     * on defaultBranch only, no need devBranch because "Deleting configuration from trash is not allowed in
+     * development branches."
+     * @return void
      */
-    public function testComponentConfigDelete($clientType)
```
Comment (martinjunger) [inline tests/Common/ComponentsTest.php:251]: Tady ten dataProvider muze zustat, pac staci jen vynechat posledni call na purge. Viz podminka na radku 304 (https://github.com/keboola/storage-api-php-client/pull/722/files?diff=split&w=1#diff-2726ab8159393b2a45bafacf5d6e3b421b0de65fd569e2f0d96339a7e0b83c8aL334)

Comment (martinjunger) [reply tests/Common/ComponentsTest.php:251]: Respektive Tomasova uprava: https://github.com/keboola/storage-api-php-client/pull/706/commits/3f849fc7cd67a37adb4058d3ca3d0ac970a129f6

Comment (martinjunger) [reply tests/Common/ComponentsTest.php:251]: Takze je tam error: `Undefined variable '$clientType' `

Comment (martinjunger) [reply tests/Common/ComponentsTest.php:251]: Jak na to koukam, to by slo sjednotit do jednoho s podminkama pro
- `purge` v `default` branchi
- `// check main branch` v `dev` branchi

Je to dobrej napad?

**Review by martinjunger — CHANGES_REQUESTED** [review]
Jeste spravit ten `ComponentsTest\testComponentConfigDelete`.
Jinak to vypada OK.

File: tests/Common/ComponentsTest.php (hunk 1 above)
Comment (jirkasemmler) [reply tests/Common/ComponentsTest.php:251]: dohodli jsme se
- mrknu jestli testComponentConfigDelete a BranchCompoenntTest::testDeleteBranchConfiguration dela cca to same
- pokud ano, tak tady to pustime bez providera a smazu podminku na konci
- zkusim tady ten test trochu okomentovat jako to ma ten test v branchi at to je trochu prehledne

File: tests/Common/ComponentsTest.php [hunk 2]
```diff
@@ -242,41 +242,33 @@ public function testComponentConfigurationDelete()
...
 
-        $listConfigurationOptions = (new ListComponentConfigurationsOptions())
-            ->setComponentId($componentId)
-            ->setIsDeleted(false);
-
-        $result = $components->listComponentConfigurations($listConfigurationOptions);
-        $this->assertCount(0, $result);
-
-        $result = $components->listComponents((new ListComponentsOptions())->setIsDeleted(false));
-        $this->assertCount(0, $result);
```
Comment (martinjunger) [inline tests/Common/ComponentsTest.php]: tohle by tu mohlo zustat, ne?

Comment (jirkasemmler) [reply tests/Common/ComponentsTest.php]: ono tyto dva asserty jsou duplicitni s asserty na lajne 264 a 265 (noveho kodu tohoto commitu)

Comment (martinjunger) [reply tests/Common/ComponentsTest.php]: aha, tady je navic jen `setIsDeleted(false)`

Comment (jirkasemmler) [reply tests/Common/ComponentsTest.php]: coz je defaultni...

Comment (martinjunger) [reply tests/Common/ComponentsTest.php]: OK

**Review by martinjunger — APPROVED** [review]
OK

**Pull request merged**